ffms2-sys/build.rs

```diff
@@ -16,11 +17,50 @@ fn format_write(builder: bindgen::Builder) -> String {
 }
 
 fn main() {
-    let libs = metadeps::probe().unwrap();
-    let headers = libs.get("ffms2").unwrap().include_paths.clone();
+    // Consider 'FFMS_INCLUDE_DIR' and 'FFMS_LIB_DIR', if pkg-config should not be used.
+    let headers = env::var("FFMS_INCLUDE_DIR").map(|value| {
+        // Ensure the include directory is valid
+        let include_dir = PathBuf::from(value.as_str());
+        if !include_dir.is_dir() {
+            panic!("The specified include directory '{}' in FFMS_INCLUDE_DIR is not valid.", value);
+        }
 
-    let mut builder = bindgen::builder().header("data/ffms.h");
+        // Get the path to the lib.
+        let lib_dir = env::var("FFMS_LIB_DIR").and_then(|lib_dir| {
+            match PathBuf::from(lib_dir) {
+                lib_dir if lib_dir.is_dir() => Ok(lib_dir),
+                _ => Err(VarError::NotPresent)
+            }
+        }).expect("FFMS_LIB_DIR is not set or the specified directory is not valid.");
+
+        // Using dynamic library in Windows remains a problem. We have to copy the DLL into a path...
+        // Problem: If 'FFMS_LIB_DIR' is outside of 'target', it is not considered (https://doc.rust-lang.org/cargo/reference/environment-variables.html).
+        #[cfg(windows)] {
+            // We need to add the file in target/{debug|release as it is included in the PATH: https://doc.rust-lang.org/cargo/reference/environment-variables.html#dynamic-library-paths
+            let output: PathBuf = [env::var("OUT_DIR").expect("Unable to get OUT_DIR").as_ref(), "..", "..", "..", "ffms2.dll"].iter().collect();
```

Comment:
Since `output` is only used to retrieve information I prefer having a `Path`, as written [here](https://doc.rust-lang.org/std/path/index.html#simple-usage).
And we can move the `env::var...` in another variable too

Reply:
Moving env out sound like a plan. But I would like to rely on `collect`. And only that only works for PathBuf: `impl<P: AsRef<Path>> FromIterator<P> for PathBuf`...

Reply:
I see, so let's use your approach but with the `env::var...` in another variable :+1: